operator/pkg/model/translation/translator.go

```diff
@@ -337,11 +337,18 @@ func getNamespaceNamePortsMap(m *model.Model) map[string]map[string][]string {
 	namespaceNamePortMap := map[string]map[string][]string{}
 	for _, l := range m.HTTP {
 		for _, r := range l.Routes {
-			for _, be := range r.Backends {
+			for bIdx, be := range r.Backends {
 				namePortMap, exist := namespaceNamePortMap[be.Namespace]
 				if exist {
 					namePortMap[be.Name] = slices.SortedUnique(append(namePortMap[be.Name], be.Port.GetPort()))
 				} else {
+					if be.Port == nil {
+						be.Port = &model.BackendPort{
+							Port: l.Port,
```

Comment:
Going by the GatewayAPI spec  https://gateway-api.sigs.k8s.io/reference/spec/#gateway.networking.k8s.io/v1.BackendObjectReference 
```
Port is required when the referent is a Kubernetes Service
```

This would mean we should invalidate a route without a port (as we only support services as backend) and they should not be allowed to end up in the translator. Setting a port to them would introduce non spec compliant behaviour. We should find out why the users was not given a rejected route when the backend port wasn't set as this seems weird to me that it did not happen.

Reply:
ah that's correct, the GatewayAPI project has added CEL validation for `Service` to must have a `port` field

Reply:
The user seems to have cilium v0.14 branch, and thus still referred to v0.7.1 Gateway API which was not enabling the CEL validation. I think we can close the PR then.

Reply:
I think we can indeed close this as fixed in upstream

Reply:
sure @meyskens